src/useRouteParam/useRouteParam.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-redeclare */
+import { computed, Ref } from 'vue'
+import { useRoute } from 'vue-router'
+
+export function useRouteParam(param: string): Ref<string | string[]>
+export function useRouteParam(param: string, defaultValue: string): Ref<string>
+export function useRouteParam(param: string, defaultValue: string[]): Ref<string[]>
+export function useRouteParam(param: string, defaultValue?: string | string[]): Ref<string | string[]> {
```

Comment:
This should not support default values. They either exist in the url or they don't.

Reply:
I wonder about the return type as well. Technically route params can be `string` or `string[]`. So technically this is correct. But I believe we've been using this assuming it's `string` and we'd have to do an additional type check every time we use this. 

I'd like to see a method for getting a param that returns `string` so we don't have to do additional type guarding. (maybe an additional method, or a `first` optional Boolean argument?)